src/Module/Admin/Site.php

```diff
@@ -60,70 +61,19 @@ protected function post(array $request = [])
 			return;
 		}
 
-		// relocate
-		// @TODO This file could benefit from moving this feature away in a Module\Admin\Relocate class for example
-		if (!empty($_POST['relocate']) && !empty($_POST['relocate_url']) && $_POST['relocate_url'] != "") {
-			$new_url = $_POST['relocate_url'];
-			$new_url = rtrim($new_url, "/");
-
-			$parsed = @parse_url($new_url);
-			if (!is_array($parsed) || empty($parsed['host']) || empty($parsed['scheme'])) {
-				notice(DI::l10n()->t("Can not parse base url. Must have at least <scheme>://<domain>"));
-				DI::baseUrl()->redirect('admin/site');
+		if (!empty($_POST['relocate']) && !empty($_POST['relocate_url'])) {
+			try {
+				$relocate = new Relocate(DI::baseUrl(), DI::dba(), DI::config(), DI::logger());
+				$relocate->run($_POST['relocate_url']);
```

Comment:
We change that much data during the execution of this command, that I don't think that we can execute it via the web frontend. It will run into a timeout and then will leave the system in a quirky state. We should only allow the relocate command via the command line.

Reply:
Ok, fair enough.

Reply:
Sorry, that I hadn't thought about this earlier.